### PR TITLE
[State Observability] Promote the API to alpha

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

- Promote APIs to PublicAPI(alpha)
- Change pre-alpha -> alpha
- Fix a bug ray_logs is displayed to `ray --help`

Release test result: https://github.com/ray-project/ray/pull/26610
Some APIs are subject to change at the beta stage (e.g., ray list jobs or ray logs).

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/experimental/state/api.py [hunk 1]
```diff
@@ -618,6 +623,7 @@ def get_node(
     )
 
 
+@PublicAPI(stability="alpha")
```
Comment (rkooo567) [inline python/ray/experimental/state/api.py:626]: Btw, should I take it out from the experimental folder? Or should I do when it becomes beta (thought the latter, but lmk if it is the former)

File: python/ray/scripts/scripts.py [hunk 2]
```diff
@@ -2563,7 +2564,7 @@ def add_command_alias(command, name, hidden):
 cli.add_command(cpp)
 cli.add_command(disable_usage_stats)
 cli.add_command(enable_usage_stats)
-add_command_alias(ray_logs, name="logs", hidden=False)
+add_command_alias(ray_logs, name="logs", hidden=True)
```
Comment (rickyyx) [inline python/ray/scripts/scripts.py:2567]: Ah nice - thanks. Was about to make a PR on this.

**Review by rickyyx — COMMENTED** [review]
<div><br/>via <a href="https://giphy.com/gifs/shaq-shimmy-UO5elnTqo4vSg">GIPHY</a></div>

File: python/ray/scripts/scripts.py (hunk 2 above)
Comment (rickyyx) [reply python/ray/scripts/scripts.py:2567]: Should I stamp this?

**Review by scv119 — APPROVED** [review]

File: doc/source/ray-observability/state/ray-state-api-reference.rst [hunk 3]
```diff
@@ -3,7 +3,9 @@ Ray State API
 
 .. _state-api-ref:
 
-.. tip:: APIs are pre-alpha and under active development. APIs are subject to change and not stable across versions.
+.. note:: 
+
```
Comment (richardliaw) [inline doc/source/ray-observability/state/ray-state-api-reference.rst:7]: can you link to the definition of alpha here (api-stability) and reduce verbosity?

File: python/ray/experimental/state/api.py (hunk 1 above)
Comment (richardliaw) [reply python/ray/experimental/state/api.py:626]: imo you should move out of experimental.

**Review by richardliaw — CHANGES_REQUESTED** [review]

**Comment by rkooo567** [comment]
So, I removed the change to annotate Python APIs as alpha as it is required to be taken out from experimental. 

This might requires some changes, and has a risk of not being picked. But we should at least mark the CLI as alpha, so let's merge this first.

File: python/ray/experimental/state/api.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/experimental/state/api.py]: https://github.com/ray-project/ray/pull/27846/files -> it will be done here.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**